### PR TITLE
feat: argocd extension for nuumarollouts

### PR DESCRIPTION
**Pull request by darshansimha**
ISB Rollout :

Screenshot 2024-07-10 at 11 12 48 AM

Controller Rollout:

Screenshot 2024-07-10 at 11 13 31 AM

Pipeline Rollout :

Screenshot 2024-07-10 at 11 13 57 AM

### CODE REVIEW

File: extensions/argo/numa-rollout/numa-rollout-extension/ui/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,12762 @@
+{
+  "name": "numarollout",
+  "version": "0.1.0",
+  "lockfileVersion": 1,
+  "requires": true,
+  "dependencies": {
+    "@adobe/css-tools": {
+      "version": "4.4.0",
+      "resolved": "https://registry.npmjs.intuit.com:443/artifactory/api/npm/npm-intuit/@adobe/css-tools/-/css-tools-4.4.0.tgz",
```
Comment (mshakira) [inline extensions/argo/numa-rollout/numa-rollout-extension/ui/package-lock.json:9]: You need to change the registry from intuit

File: extensions/argo/numa-rollout/numa-rollout-extension/ui/src/rollout/ControllerRollout.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  }, [props.tree]);
+  const controllerPods = useMemo(() => {
+    if (!kindToNodeMap || !kindToNodeMap.get("Pod")) {
+      return [];
+    }
+    const pods = kindToNodeMap.get("Pod");
+    if (!pods) {
+      return [];
+    }
+    return pods.filter((node) => node.name.indexOf("numaflow-controller") >= 0);
```
Comment (mshakira) [inline extensions/argo/numa-rollout/numa-rollout-extension/ui/src/rollout/ControllerRollout.tsx:38]: include null check for node.name

File: extensions/argo/numa-rollout/numa-rollout-extension/ui/src/rollout/PipelineRollout.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+                  backgroundColor: "#dee6eb",
+                  borderRadius: "3px",
+                  padding: "6px",
+                  flexWrap: "wrap",
+                  marginRight: "-3px",
+                  marginBottom: "-3px",
+                }}
+              >
+                Pipeline Status:{" "}
+                {props.resource.status.conditions[0].type ===
```
Comment (mshakira) [inline extensions/argo/numa-rollout/numa-rollout-extension/ui/src/rollout/PipelineRollout.tsx:51]: null check

File: extensions/argo/numa-rollout/numa-rollout-extension/ui/src/rollout/default/ArgoRolloutComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+      rolloutAnalysisRunInfo();
+    }
+  }, [isRefresh, analysisRunNodeIds]);
+
+  return analysisRunResults;
+};
+
+const parseAnalysisRunStatus = (status: string): string => {
+  switch (status) {
+    case "Healthy":
```
Comment (mshakira) [inline extensions/argo/numa-rollout/numa-rollout-extension/ui/src/rollout/default/ArgoRolloutComponent.tsx:202]: can we move all string to variables? like `const HEALTHY = "Healthy"

Comment (darshansimha) [reply extensions/argo/numa-rollout/numa-rollout-extension/ui/src/rollout/default/ArgoRolloutComponent.tsx]: Done!

**Pull request closed without merging**